internal/storage/storage_handle.go

```diff
@@ -13,7 +13,11 @@ import (
 	"google.golang.org/api/option"
 )
 
-type storageClient struct {
+type StorageHandle interface {
+	BucketHandle(bucketName string) (bh *bucketHandle, err error)
+}
+
+type StorageClient struct {
```

Comment:
why is this made public?

Reply:
Got a warning that return type of a Exported (public) method should also be public. I can revert if it doesn't make sense.

Reply:
you can return an interface type here instead of structType

Reply:
Like this: https://github.com/jacobsa/gcloud/blob/8d1a66f33ec963197b578d3bd491df1f6deea91b/gcs/bucket.go#L352